thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp

```diff
@@ -1379,7 +1422,7 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor, implem
         }
         virtual void process(CThorExpandingRowArray &processing, unsigned __unused) override
```

Comment:
Am I correct in understanding that process() does not have to be thread safe?

Reply:
It was thread safe, as there was only 1 thread in process() (Each handler has 1 process() thread,) and all the entities it dealt with were either it's own, pre-intiialized, or made thread safe on access.
The introduction of another thread that interacted with things the process() thread initializes meant there was potential race conditions that needed to be protected from.